webapp/components/navbar.jsx

```diff
@@ -187,18 +195,29 @@ export default class Navbar extends React.Component {
             const curIndex = allChannels.indexOf(curChannel);
             let nextChannel = curChannel;
             let nextIndex = curIndex;
+            let count = 0;
             if (e.keyCode === Constants.KeyCodes.UP) {
-                while (nextIndex >= 0 && ChannelStore.getUnreadCount(allChannels[nextIndex].id).msgs === 0 && ChannelStore.getUnreadCount(allChannels[nextIndex].id).mentions === 0) {
+                while (count < allChannels.length && nextIndex >= 0 && ChannelStore.getUnreadCount(allChannels[nextIndex].id).msgs === 0 && ChannelStore.getUnreadCount(allChannels[nextIndex].id).mentions === 0) {
                     nextIndex--;
+                    count++;
+                    if (nextIndex < 0) {
+                        nextIndex = allChannels.length - 1;
+                    }
                 }
             } else if (e.keyCode === Constants.KeyCodes.DOWN) {
-                while (nextIndex <= allChannels.length - 1 && ChannelStore.getUnreadCount(allChannels[nextIndex].id).msgs === 0 && ChannelStore.getUnreadCount(allChannels[nextIndex].id).mentions === 0) {
+                while (count < allChannels.length && nextIndex <= allChannels.length - 1 && ChannelStore.getUnreadCount(allChannels[nextIndex].id).msgs === 0 && ChannelStore.getUnreadCount(allChannels[nextIndex].id).mentions === 0) {
                     nextIndex++;
+                    count++;
+                    if (nextIndex >= allChannels.length) {
+                        nextIndex = 0;
+                    }
                 }
             }
-            if (nextIndex !== curIndex && ChannelStore.getUnreadCount(allChannels[nextIndex].id).msgs !== 0 || ChannelStore.getUnreadCount(allChannels[nextIndex].id).mentions !== 0) {
-                nextChannel = allChannels[nextIndex];
-                GlobalActions.emitChannelClickEvent(nextChannel);
+            if (nextIndex >= 0 && nextIndex < allChannels.length) {
```

Comment:
Isn't this check a bit redundant? Won't `nextIndex` always be `>= 0` and `< allChannels.length` at this point due to the conditions you added above?